gateway/src/main/java/com/mx/path/gateway/util/UpstreamLogger.java

```diff
@@ -90,7 +90,7 @@ public final void logRequest(Response response) {
 
     Request request = response.getRequest();
     MDC.put("request_method", request.getMethod());
-    MDC.put("request_uri", request.getUri());
```

Comment:
![Screenshot 2023-12-15 at 2 42 14 PM](https://github.com/mxenabled/path-core/assets/2914335/1165590f-dae7-481c-9421-35c18624538d)
Are you wanting to mask this? This is a very commonly used field when we are debugging upstream errors on third parties

Reply:
This is not common. Is there something sensitive in the URI?

Reply:
I'm open to discuss options. This is coming from FCTI origination They have a URI:

https://fctimx.okta.com/api/v1/users?filter=profile.login+eq+%22john.doe1%40icloud.com%22&limit=1

If we don't mask this then the username john.doe@icloud.com is in the logs.

Reply:
In my testing the logs now show:
https://clientX.okta.com/api/v1/users?filter=profile.login+eq+%22**MASKED**%22&limit=1

Reply:
What regex is it using to limit the string to just that

Reply:
The connector is adding two:

LogValueMasker.registerPayloadPattern("users[\\?&].*q=([^&]*)&?");
LogValueMasker.registerPayloadPattern("users[\\?&].*filter=profile\\.login\\+eq\\+%22([^&]*)%22&?");

Reply:
I do think it's odd, however, to add a payload pattern for the URI. Maybe this could warrant .registerURIPattern?

Reply:
Or rename registerPayloadPattern to registerMaskPattern.

Reply:
I think having a separate URI pattern set would be good that way something does not get caught that starts masking to much

Reply:
It doesn't seem likely that we would accidentally mask important parts of the request URI. Our masking patterns are usually pretty specific (as they should be). Separate making patterns for URIs is probably overkill IMO, but it would prevent unintended URI masking.